tools/firtool/firtool.cpp

```diff
@@ -360,6 +360,25 @@ static cl::opt<bool> stripDebugInfo(
     cl::desc("Disable source locator information in output Verilog"),
     cl::init(false), cl::cat(mainCategory));
 
+// Build mode options.
+enum BuildMode { BuildModeDebug, BuildModeRelease };
+static cl::opt<BuildMode> buildMode(
+    "O", cl::desc("Controls how much optimization should be performed."),
+    cl::values(clEnumValN(BuildModeDebug, "debug",
+                          "compile with only necessary optimizations"),
+               clEnumValN(BuildModeRelease, "release",
+                          "compile with optimizations")),
+    cl::init(BuildModeRelease), cl::callback([](const BuildMode &buildMode) {
+      switch (buildMode) {
+      case BuildModeDebug:
+        preserveMode = firrtl::PreserveValues::Named;
```

Comment:
It would be super neat if whatever set of options are set are things that a user could see from `--help` or something.  That is, they could see that `-O debug` sets `preserveMode=None`.  I guess, documentation or just look at the source?  Especially since these possibly interact with user controlled options, it'd be good to know what they entail I think (and informative when deciding which to use).

Aww, `--print-options` and `--print-all-options` kinda address this need, although they don't appear to work for firtool 😢 .

Other than those options I'm not sure how to do that without a bunch more code but just wanted to mention it and put it out there.

Reply:
:( not sure how to get something like this working.